doc/GUIDE.md

```diff
@@ -280,6 +280,23 @@ As you can see from that path (and as emphasized earlier), the installation is
 placed to not interfere with any other GHC installation, whether system-wide or
 even different GHC versions installed by stack.
 
+## Cleaning your project
+You can clean up build artifacts for your project using the `stack clean` and `stack purge`.
+
+### `stack clean`
+`stack clean` Deletes the local working directories containing compiler output. 
+(typically '.stack-work/dist' by default)
+
+Use `stack clean <specific-package>` to delete the output for a package specific-package only.
+
+### `stack purge`
+`stack purge` Deletes the local stack working directory, including extra-deps, git dependencies and the compiler output. 
+It does not delete any snapshot packages, compilers or installed programs. -- This essentially
+reverts your project to a completely fresh state, as if it had never been built. 
```

Comment:
This last bit could be more precise. It isn't obvious what a 'completely fresh state' is. I'd have to study that part of the code to offer something that could be written here instead.

I suspect the list I wrote in my comment on the issue may not be comprehensive, so I'll get somebody else to check.